### PR TITLE
Reformat meshgrid

### PR DESCRIPTION
**Pull request by dianarad**
This PR 

- updates function args and type hints for the meshgrid function
- touches up on spacing/missing positional arg

Issue: https://github.com/unifyai/ivy/issues/7496

### CODE REVIEW

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by dianarad** [comment]
still working on PR (just to remove the stale label)

**Review by simonetgordon — CHANGES_REQUESTED** [review]

File: ivy/functional/ivy/creation.py
Comment (simonetgordon) [reply ivy/functional/ivy/creation.py:1158]: Hi @dianarad, apologies for the delay, my comments were not visible to you as I didn't submit the review properly. The PR looks good - just small formatting changes to be made.

Comment (dianarad) [reply ivy/functional/ivy/creation.py]: Hey @simonetgordon! No worries. I've aligned the elements, let me know if there is anything else I can do.

Comment (simonetgordon) [reply ivy/functional/ivy/creation.py]: Hi @dianarad, meshgrid is now failing on all the backends on this PR, whereas it isn't on master. You can view the stack trace on the check `test-core-ivy / run-nightly-tests (numpy, creation) (pull_request)` in `Details -> Run Functronal-Core tests`. The following is the error stack trace for numpy test_meshgridL

```
=================================== FAILURES ===================================
_________ test_meshgrid[cpu-ivy.functional.backends.numpy-False-False] _________
ivy_tests/test_ivy/test_functional/test_core/test_creation.py:512: in test_meshgrid
    fn_tree="functional.ivy.meshgrid",
ivy_tests/test_ivy/test_functional/test_core/test_creation.py:543: in test_meshgrid
    helpers.test_function(
ivy_tests/test_ivy/helpers/function_testing.py:2[68](https://github.com/unifyai/ivy/actions/runs/3744161318/jobs/6357213805#step:7:69): in test_function
    assert ivy.is_array(test_ret)
E   AssertionError
E   Falsifying example: test_meshgrid(
E       dtype_and_arrays=(['uint16', 'uint16'],
E        [array([0], dtype=uint16), array([0], dtype=uint16)]),
E       sparse=False,
E       indexing='xy',
E       test_flags=<ivy_tests.test_ivy.helpers.test_parameter_flags.FunctionTestFlags at 0x7f60e4f1e3d0>,
E       backend_fw=<module 'ivy.functional.backends.numpy' from '/ivy/ivy/functional/backends/numpy/__init__.py'>,
E       fn_name='meshgrid',
E       on_device='cpu',
E       ground_truth_backend='tensorflow',
E   )
E   
E   You can reproduce this example by temporarily adding @reproduce_failure('6.55.0', b'AXicY2BABoxADAAAFgAC') as a decorator on your test case
-----------------------
```
Please try to investigate and rectify the cause of this issue. If you're having problems debugging, please feel free to reach out to me for further guidance :)

Comment (dianarad) [reply ivy/functional/ivy/creation.py]: Hi @simonetgordon! I looked into this issue and I'm having some trouble finding the exact cause. I believe it's not related to this PR in particular, as
- I tried running this test locally, on the updated master branch of the unify (not forked) repo, which failed with the same error stack trace
- I also found [this run from a couple days ago](https://github.com/unifyai/ivy/actions/runs/3763800463/jobs/6397625331?fbclid=IwAR1YOiHWkrrmOLa7INNO_GOBqyf7STE_IB8HuJjsPIqYIB6rx5uUyy84Kjo) which shows the same error on the master branch 
- I noticed some of the passed tests on the master branch seem to just be skipped ([an example](https://github.com/unifyai/ivy/actions/runs/3777431185/jobs/6421259593))

I was curious so I looked into the code further and the error is thrown by `helpers.test_function` on a call of `meshgrid(ivy.array([0]), ivy.array([0]), sparse=False, indexing='xy')` because the output `[ivy.array([[0]]), ivy.array([[0]])]` is not an array. My understanding is that it is correct for `meshgrid` to return a list of arrays, rather than an array (for reference, the example calls in the meshgrid docs would fail this check too), so I'm not sure if the fault is with the function itself or rather the test.  

Any thoughts?

Comment (simonetgordon) [reply ivy/functional/ivy/creation.py]: @dianarad 

Thanks for diving into this issue!  Indeed, it seems to be an issue on master, so I can merge this PR once you have resolved the merge conflicts :)

Regarding the CI skips, this is intentional in some circumstances, so no worries about that.

You seem to be correct about meshgrid and since you have gone to the trouble of locating the issue, feel free to open a PR on the issue with a suggested fix so you can receive credit for tracking it down. If you don't wish to do this, let me know and I can try fix the issue 😊

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

File: ivy/functional/ivy/creation.py
Comment (dianarad) [reply ivy/functional/ivy/creation.py]: @simonetgordon 

No problem! Merge conflicts should be resolved now. The new changes also seem to have fixed the issue with the test, so someone got there with the fix before us 😄 
Thank you!

**Pull request merged**
